### PR TITLE
fix: pvc mounted on pods for cache handling

### PR DESCRIPTION
**Pull request by prakash100198**
For shorted build time, PVC has been added in pods at app level and pipeline level

### CODE REVIEW

File: helper/DockerHelper.go [hunk 1]
```diff
@@ -200,9 +200,15 @@ func BuildArtifact(ciRequest *CiRequest) (string, error) {
 	}
 	if ciBuildConfig.CiBuildType == SELF_DOCKERFILE_BUILD_TYPE || ciBuildConfig.CiBuildType == MANAGED_DOCKERFILE_BUILD_TYPE {
 		dockerBuild := "docker build "
+		if ciRequest.CacheInvalidate && ciRequest.IsPvcMounted {
+			dockerBuild = "docker build --no-cache"
```
Comment (kripanshdevtron) [inline helper/DockerHelper.go:204]: append no chache to dockerBuild instead of assigning value to it

File: util/CommonConstants.go [hunk 2]
```diff
@@ -27,6 +27,8 @@ const RETRYCOUNT = 10
 const WORKINGDIR = "/devtroncd"
 const LOCAL_BUILDX_LOCATION = "/var/lib/devtron/buildx"
 const LOCAL_BUILDX_CACHE_LOCATION = LOCAL_BUILDX_LOCATION + "/cache"
+const BUILD_CACHE_FOLDER_PVC = "/devtroncd-cache"
```
Comment (kripanshdevtron) [inline util/CommonConstants.go:30]: do we need these constants now ??

File: helper/DockerHelper.go (hunk 1 above)
Comment (prakash100198) [reply helper/DockerHelper.go]: done

File: util/CommonConstants.go (hunk 2 above)
Comment (prakash100198) [reply util/CommonConstants.go]: sorry forgot to remove them, removed now since we don't need them

**Review by kripanshdevtron — CHANGES_REQUESTED** [review]
put this check here

**Review by kripanshdevtron — APPROVED** [review]

**Pull request merged**